### PR TITLE
Dot Nonsense: find the current running proc's type!

### PR DESCRIPTION
**Pull request by RemieRichards**
This is useful, if a little insane.

### CODE REVIEW

File: code/__DEFINES/misc.dm [hunk 1]
```diff
@@ -4,6 +4,19 @@
...
 // #define WEST 8
 
+//These get to go at the top, because they're special
+//You can use these defines to get the typepath of the currently running proc/verb (yes procs + verbs are objects)
+/* eg:
+/mob/living/carbon/human/death()
+	world << THIS_PROC_TYPE_STR //You can only output the string versions
+Will print: "/mob/living/carbon/human/death" (you can optionally + "()" to make it look nicer)
+*/
+#define THIS_PROC_TYPE .....
```
Comment (ChangelingRain) [inline code/__DEFINES/misc.dm]: why is byond absolutely insane half the time

Comment (RemieRichards) [inline code/__DEFINES/misc.dm]: Don't got an answer for that one I'm afraid!

**Comment by Cyberboss** [comment]
Good lord...

**Comment by Iamgoofball** [comment]
what the literal fuck

**Comment by kevinz000** [comment]
Holy
Shit.

**Comment by Haevacht** [comment]
CYBORG!

CYYYBOOOOORG.

**Comment by RemieRichards** [comment]
I think you mean 'Good lord.....' @Cyberboss

**Comment by XDTM** [comment]
.....

**Pull request merged**